python/gstaichi/lang/ast/ast_transformers/function_def_transformer.py

```diff
@@ -44,7 +44,9 @@ def _decl_and_create_variable(
             ctx.kernel_args.append(name)
         if annotation == annotations.template or isinstance(annotation, annotations.template):
             assert ctx.global_vars is not None
-            return True, ctx.global_vars[name]
+            if name in ctx.template_vars:
+                return True, ctx.template_vars[name]
+            return True, ctx.global_vars.get(name, ctx.template_vars.get(name))
```

Comment:
You already know that `name not in ctx.template_vars`, which means that `ctx.template_vars.get(name)` evaluate to `None` systematically?